pennylane_lightning/lightning_qubit/_adjoint_jacobian.py

```diff
@@ -235,98 +108,19 @@ def calculate_jacobian(self, tape: QuantumTape):
             processed_data["ops_serialized"],
             trainable_params,
         )
+
         jac = np.array(jac)
-        has_shape0 = bool(len(jac))
+        has_shape0: bool = len(jac) > 0
```

Comment:
We do not usually use type annotations outside function interfaces, but I don't mind if we start doing it.
```suggestion
        has_shape0 = len(jac) > 0
```

Reply:
Yes, I added it because I remember that It was confusing at some point but I think it is better to keep simple